build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/info/BuildParameterExtension.java

```diff
@@ -67,4 +67,6 @@ public interface BuildParameterExtension {
     Random getRandom();
 
     Boolean isGraalVmRuntime();
+
+    void withMinimumJavaRuntimeVersion(JavaVersion version, Runnable runnable);
```

Comment:
Forgive my bikeshedding: we use the term "minimum runtime version" to mean something slightly different. I also wonder if the name here should give some clue as to the fact that the given `Runnable` is conditional. Some kind of "if" or "when" prefix might better. Perhaps `ifJavaRuntimeVersion` or `ifJavaRuntimeVersionCompatibleWith`?